native/src/main/java/io/ballerina/lib/ibm.ibmmq/CommonUtils.java

```diff
@@ -18,23 +18,180 @@
+    private static final String BPROPERTY = "Property";
+    private static final String BMESSAGE_NAME = "Message";
+    private static final BString PD_VERSION = StringUtils.fromString("version");
+    private static final BString PD_COPY_OPTIONS = StringUtils.fromString("copyOptions");
+    private static final BString PD_OPTIONS = StringUtils.fromString("options");
+    private static final BString PD_SUPPORT = StringUtils.fromString("support");
+    private static final BString PD_CONTEXT = StringUtils.fromString("context");
+    private static final BString PROPERTY_VALUE = StringUtils.fromString("value");
+    private static final BString PROPERTY_DESCRIPTOR = StringUtils.fromString("descriptor");
+
+    private static final MQPropertyDescriptor defaultPropertyDescriptor = new MQPropertyDescriptor();
+
+    public static MQMessage getMqMessageFromBMessage(BMap<BString, Object> bMessage) {
+        byte[] payload = bMessage.getArrayValue(MESSAGE_PAYLOAD).getBytes();
+        MQMessage mqMessage = new MQMessage();
+        try {
+            mqMessage.write(payload);
+        } catch (IOException e) {
+            throw createError(IBMMQ_ERROR,
+                    String.format("Error occurred while populating payload: %s", e.getMessage()), e);
+        }
+        BMap<BString, Object> properties = (BMap<BString, Object>) bMessage.getMapValue(MESSAGE_PROPERTIES);
+        populateMQProperties(properties, mqMessage);
+        return mqMessage;
+    }
+
+    public static BMap<BString, Object> getBMessageFromMQMessage(MQMessage mqMessage) {
+        BMap<BString, Object> bMessage = ValueCreator.createRecordValue(getModule(), BMESSAGE_NAME);
+        try {
+            byte[] payload = new byte[mqMessage.getDataLength()];
+            mqMessage.readFully(payload);
+            bMessage.put(MESSAGE_PAYLOAD, payload);
+            bMessage.put(MESSAGE_PROPERTY, getBProperties(mqMessage));
+            return bMessage;
+        } catch (MQException | IOException e) {
+            throw createError(IBMMQ_ERROR,
+                    String.format("Error occurred while reading the message: %s", e.getMessage()), e);
+        }
+    }
+
+    private static BMap<BString, Object> getBProperties(MQMessage mqMessage) throws MQException {
+        BMap<BString, Object> properties = ValueCreator.createMapValue(TypeCreator
+                .createMapType(TypeCreator.createRecordType(BPROPERTY, getModule(), 0, false, 0)));
+        Enumeration<String> propertyNames = mqMessage.getPropertyNames("%");
+        for (String propertyName : Collections.list(propertyNames)) {
+            BMap<BString, Object> property = ValueCreator.createRecordValue(getModule(), BPROPERTY);
+            MQPropertyDescriptor propertyDescriptor = new MQPropertyDescriptor();
+            Object propertyObject = mqMessage.getObjectProperty(propertyName, propertyDescriptor);
+            if (propertyObject instanceof Integer) {
```

Comment:
Shall we update this code according to following comment ?
https://github.com/ballerina-platform/module-ballerinax-ibm.ibmmq/pull/3#discussion_r1374554585